subsys/net/lib/openthread/platform/radio.c

```diff
@@ -361,12 +362,49 @@ otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
 	return &sTransmitFrame;
 }
 
+static void get_rssi_energy_detected(struct device *dev, s16_t max_ed)
+{
+	ARG_UNUSED(dev);
+	ot_signal_set(OPENTHREAD_SIGNAL_BLOCKING, max_ed);
+}
+
 int8_t otPlatRadioGetRssi(otInstance *aInstance)
 {
-	ARG_UNUSED(aInstance);
+	s8_t ret_rssi = INT8_MAX;
+	int signal_value;
+
+	otRadioCaps radioCaps =  otPlatRadioGetCaps(aInstance);
+
+	if (!(radioCaps & OT_RADIO_CAPS_ENERGY_SCAN)) {
+		/*
+		 *TODO: No API in Zephyr to get the RSSI
+		 * when OT_RADIO_CAPS_ENERGY_SCAN is not available
+		 */
+		ret_rssi = 0;
+	} else {
+		/*
+		 * Blocking implementation of get RSSI
+		 * using no-blocking ed_scan
+		 */
+		int error = 0;
+		const u16_t energy_detection_time = 1;
+
+		error = radio_api->ed_scan(radio_dev, energy_detection_time,
```

Comment:
Given [this commit](https://github.com/openthread/openthread/commit/1edde90d91fff21343bdb8560eafcba6d5cf1331) I have serious concerns about implementing otPlatRadioGetRssi on top of Energy Scan. Apparently, this is not a configuration that is supported by OpenThread (see assert).

Reply:
Me too.
But our sprint process required that implementation, and it was decided to do that way.
But, lets be calm. That does need to be added to Zephyr. Now the most important is to define optimal shim interface.